### ISSUE TITLE
Создать ветку из другого хранилища

### ISSUE THREAD

**Issue by KeyFire** [1C-Company/GitConverter#79]
**Существует проблема**
Раз в 2-3 месяца требуется создать релизное хранилище и в мастере поднять версию. При этом исправления ошибок делаются в релизном хранилище. А новые задачи заливаются в master.
Так как операция редкая, забываешь, что делаешь. Требуется автоматизация действий:
1. Идем в git, выбираем коммит, из которого будем создавать бранч 
2. Создаем Бранч по выбранному коммиту 
3. Добавляем новые параметры подключения к хранилищу бранча в CI 
4. Меняем в файле созданного бранча .gitlab-ci.yml строчку c подключением к хранилищу.
5. Идем на сервер гитлаба, куда делаются выгрузки, создаем папки: dumps-release	branch-release
6. Заходим в git и делаем clone созданного бранча в созданную папку branch-release 
7. Заходим в склонированную папку и в ней в параметрах репозитория прописываем параметры подключения к GitLab и еще пару команд
8. Заходим в Гитконвертер, добавляем новое хранилище
9. Добавляем пару воркеров для загрузки метаданных и выгрузки конфигурации "Выгрузка конфигурации" и "Загрузка метаданных".

**Хотелось бы иметь следующую функциональность**
Сделать кнопку "Создать ветку из другого хранилища" которая будет выполнять все действия по клонированию и настройке.

**Расширение или улучшение?**

Это улучшение конфигурации 1С:ГитКонвертер потому что появится новая полезная функциональность..

Это функциональность можно реализовать в расширении.

**Дополнительный контекст**
Любая дополнительная информация

**Comment by marmyshev** [comment]
Кнопку, открывающую такой "помощник" можно сделать в "версии хранилища" - там есть информация о хеше коммита.
Имхо, п.4 надо делать в каком-то расширении после выполнения всех остальных действий по настройке и клонированию, т.к. это завязано на конкретный сервер CI и какой-то процесс внутри этого CI.
Все остальные пункты нужно делать всем...

**Comment by Zavailik1365** [comment]
> 7\. Заходим в склонированную папку и в ней в параметрах репозитория прописываем параметры подключения к GitLab и еще пару команд

Это тоже будет в помощнике? Кажется, нужно выносить в расширение.

**Comment by marmyshev** [comment]
@KeyFire  по п.7 наверное имелось ввиду "парочка команд" - это то как настраивается репозиторий для EDT через `git config --local ...` верно?

**Comment by mrshadow300373** [comment]
Могу порекомендовать по это ише мой форк. https://github.com/mrshadow300373/GitConverter В принципе, как я понял он именно это и делает, правда я его для другого делал.

**Comment by marmyshev** [comment]
Да, в принципе идея выглядит похоже. Посмотрите актуальный master проекта - начальная инициализация существенно доработана.

Текущий тикет - скорее про создание новой ветки на основе какой-то версии хранилища... надо сделать чтобы эта операция делалась быстро и удобно - выбрал версию, задал имя ветки  и всё (в минимально варианте).

**Comment by mrshadow300373** [comment]
_делалась быстро и удобно - выбрал версию, задал имя ветки и всё (в минимально варианте)._
Так и делается, если включена галка коммитить в  существующий репо, то кнопка инициировать репо в форме скрывается, открывается другая,  которая клонирует и дополняет репозиторий необходимыми настройками (аналогично тем, что делаются при инициализации пустого каталога).См. пикчу [http://xn--e1agfaq7azal4a.xn--p1ai/download/1606755607680.png](url) Единственный минус, при открытии формы проверка каталога длится дольше, гораздо, чем при обычной настройке.
